src/systems/filecoin_blockchain/storage_power_consensus/storage_power_actor.go

```diff
@@ -347,10 +344,10 @@ func (spa *StoragePowerActor_I) Surprise(ticket block.Ticket) []addr.Address {
 	// var currBlockHeight util.UInt
 
 	// The number of miners that are challenged at this block
-	challengeCount := uint(len(spa.PowerTable())) / provingPeriod
+	challengeCount := uint(len(st.PowerTable())) / provingPeriod
```

Comment:
move the `h, st := a.State(rt)` line to right above this. (wrap the state look-ups tightly)